Feature/change timeout logic

cshcomcom opened the pull request:
Done
- Delete blocking timed out transaction logic (executeTransactionAndAddToPool)
- Use received_at to delete old transactions

platfowner commented on tx-pool/index.js:
```diff
@@ -48,6 +48,7 @@ class TransactionPool {
     if (!(tx.address in this.transactions)) {
       this.transactions[tx.address] = [];
     }
+    tx.received_at = Date.now();
```
I think it's not a good idea to change tx itself?

platfowner replied:
Instead, what about adding 'created_at' field here?
https://github.com/ainblockchain/ain-blockchain/blob/develop/tx-pool/transaction.js#L16

cshcomcom replied:
Looks good to me! I work it and open this pr again

cshcomcom commented:
@platfowner All comments are addressed! PTAL

liayoo reviewed (approved):
LGTM, thanks! Please wait for @platfowner's review.

platfowner commented on node/index.js:
```diff
@@ -234,7 +234,7 @@ class BlockchainNode {
         }
       }
-      return { tx_list: txList };
+      return {tx_list: txList};
```
minor: Personally I see that the previous pattern is more readable.

platfowner commented on tx-pool/index.js:
```diff
@@ -56,6 +56,7 @@ class TransactionPool {
       timestamp: tx.tx_body.timestamp,
       is_finalized: false,
       finalized_at: -1,
+      tracked_at: Date.now(),
```
`tracked_at: tx.created_at,` ?

cshcomcom replied:
Good! Done

platfowner commented on tx-pool/transaction.js:
```diff
@@ -1,18 +1,19 @@
 const ainUtil = require('@ainblockchain/ain-util');
 const logger = require('../logger')('TRANSACTION');
-const { WriteDbOperations } = require('../common/constants');
+const {WriteDbOperations} = require('../common/constants');
```
ditto

cshcomcom commented:
PTAL! I deleted object-curly-spacing according to the google style guide and also changed lint. I think we have to discuss coding convention and change lint before add object-curly-spacing! @platfowner

platfowner commented:
@cshcomcom  what about just turning the option off until we get any agreement so that authors can choose depending their own preference?
https://eslint.org/docs/rules/object-curly-spacing#when-not-to-use-it

cshcomcom commented:
Good idea. I turned off option and reverted it! Done

platfowner commented on tx-pool/transaction.js:
```diff
@@ -201,7 +202,7 @@ class Transaction {
 
   static isValidFormat(txBody) {
     const sanitized = Transaction.sanitizeTxBody(txBody);
-    return _.isEqual(JSON.parse(JSON.stringify(sanitized)), txBody, { strict: true });
```
ditto

platfowner reviewed (approved):
LGTM
w/ some minor comments.
Thanks for the upgrade!

cshcomcom commented:
Thanks for the review! Now merging
